SingularityBase/src/main/java/com/hubspot/singularity/TaskCleanupType.java

```diff
@@ -3,8 +3,8 @@
 public enum TaskCleanupType {
 
   USER_REQUESTED(true, true), USER_REQUESTED_TASK_BOUNCE(false, false), DECOMISSIONING(false, false), SCALING_DOWN(true, false), BOUNCING(false, false), INCREMENTAL_BOUNCE(false, false),
-  DEPLOY_FAILED(true, true), NEW_DEPLOY_SUCCEEDED(true, false), DEPLOY_STEP_FINISHED(true, false), DEPLOY_CANCELED(true, true), UNHEALTHY_NEW_TASK(true, true),
-  OVERDUE_NEW_TASK(true, true), USER_REQUESTED_DESTROY(true, true), INCREMENTAL_DEPLOY_FAILED(false, true), INCREMENTAL_DEPLOY_CANCELLED(false, true), PRIORITY_KILL(true, true);
+  DEPLOY_FAILED(true, true), NEW_DEPLOY_SUCCEEDED(true, false), DEPLOY_STEP_FINISHED(true, false), DEPLOY_CANCELED(true, true), DEPLOY_TIMED_OUT(true, true), UNHEALTHY_NEW_TASK(true, true),
```

Comment:
I'm not sure if `DEPLOY_TIMED_OUT` is the right name for this. The "DEPLOY" has the connotation that a deployment took too long (i.e. it took too long for a task to become healthy). `TASK_EXCEEDED_TIME_LIMIT` might be a better option (open to other ideas too)